src/Intervals.jl

```diff
@@ -18,23 +19,52 @@ struct Unbounded <: Bound end
 
 bound_type(x::Bool) = x ? Closed : Open
 
-abstract type AbstractInterval{T, L <: Bound, R <: Bound} end
+# Methods for convert between int and tuple representations for space efficiency
+# (Open, Closed) is 16 bytes while the integer represenation is only 1 byte.
+# TODO: Convert types to symbols when we switch to extending IntervalSets.jl
+bounds_int(l::Type{Open}, r::Type{Open}) = 0x00
+bounds_int(l::Type{Closed}, r::Type{Open}) = 0x01
+bounds_int(l::Type{Open}, r::Type{Closed}) = 0x02
+bounds_int(l::Type{Closed}, r::Type{Closed}) = 0x03
+
+bounds_types(x::Integer) = bounds_types(Val(UInt8(x)))
```

Comment:
Missed opportunities to use bit masks:
```suggestion
bounds_int(l::Type{Open}, r::Type{Open}) = 0x00 | 0x00
bounds_int(l::Type{Closed}, r::Type{Open}) = 0x10 | 0x00
bounds_int(l::Type{Open}, r::Type{Closed}) = 0x00 | 0x01 
bounds_int(l::Type{Closed}, r::Type{Closed}) = 0x10 | 0x01

function bounds_types(x::UInt8)
    L = if x & 0x20 == 0x20
        Unbounded
    elseif x & 0x10 == 0x10
        Closed
    else
        Open
    end
    
    R = if x & 0x02 == 0x02
        Unbounded
    elseif x & 0x01 == 0x01
        Closed
    else
        Open
    end
end
```
Can make this logic even cleaner with something like: https://github.com/JuliaTime/TimeZones.jl/blob/master/src/class.jl